src/dist/manifestation/tests.rs

```diff
@@ -388,108 +349,28 @@ fn rename_component_new() {
-            change_channel_date(url, "nightly", "2016-02-02");
-            run_future(update_from_dist(
-                url,
-                toolchain,
-                prefix,
-                &adds,
-                &[],
-                download_cfg,
-                tmp_cx,
-                false,
-            ))
-            .unwrap();
-            // As a result `bin/bonus` is present but not `bin/bobo` which we'd
-            // expect since the bonus component installs `bin/bonus` regardless of
-            // its name being `bobo`
-            assert!(!utils::path_exists(prefix.path().join("bin/bobo")));
-            assert!(utils::path_exists(prefix.path().join("bin/bonus")));
-        },
-    );
-}
-
-// Installs or updates a toolchain from a dist server.  If an initial
-// install then it will be installed with the default components.  If
-// an upgrade then all the existing components will be upgraded.
-// FIXME: Unify this with dist::update_from_dist
-#[allow(clippy::too_many_arguments)]
-async fn update_from_dist(
-    dist_server: &Url,
-    toolchain: &ToolchainDesc,
-    prefix: &InstallPrefix,
-    add: &[Component],
-    remove: &[Component],
-    download_cfg: &DownloadCfg<'_>,
-    tmp_cx: &temp::Context,
-    force: bool,
-) -> Result<UpdateStatus> {
-    // Download the dist manifest and place it into the installation prefix
-    let manifest_url = make_manifest_url(dist_server, toolchain)?;
-    let manifest_file = tmp_cx.new_file()?;
-    utils::download_file(
-        &manifest_url,
-        &manifest_file,
-        None,
-        &|_| {},
-        download_cfg.process,
-    )
-    .await?;
-    let manifest_str = utils::read_file("manifest", &manifest_file)?;
-    let manifest = Manifest::parse(&manifest_str)?;
-
-    // Read the manifest to update the components
-    let trip = toolchain.target.clone();
-    let manifestation = Manifestation::open(prefix.clone(), trip.clone())?;
-
-    // TODO on install, need to add profile components (but I guess we shouldn't test that logic here)
-    let mut profile_components = manifest.get_profile_components(Profile::Default, &trip)?;
-    let mut add_components = add.to_owned();
-    add_components.append(&mut profile_components);
-
-    let changes = Changes {
-        explicit_add_components: add_components,
-        remove_components: remove.to_owned(),
-    };
-
-    manifestation
-        .update(
-            &manifest,
-            changes,
-            force,
-            download_cfg,
-            &toolchain.manifest_name(),
-            true,
-        )
-        .await
+    let cx = TestContext::from_dist_server(mock_dist_server, url, GZOnly);
+
+    let adds = [Component::new(
+        "bobo".to_string(),
+        Some(TargetTriple::new("x86_64-apple-darwin")),
+        true,
+    )];
+    // Install the basics from day 1
+    change_channel_date(&cx.url, "nightly", "2016-02-01");
+    cx.update_from_dist(&[], &[], false).await.unwrap();
```

Comment:
Probably not worth doing this now, but in the future, consider splitting the commits by first isolating the `TestContext` type to contain all the arguments given to these closures (generally by the same name) and passing the `TestContext` type to the testing closures as before (and perhaps also attaching the `update_from_dist()` function to that `TestContext` type). That would have made this a lot easier to review.